src/app/pie-loan/pie-loan.component.ts

```diff
@@ -0,0 +1,59 @@
+import {Component, Input, OnInit} from '@angular/core';
+import {Observable} from "rxjs";
+import {MonthlyInterest} from "../interfaces/monthlyInterest";
+import {Chart} from 'chart.js/auto';
+import {color} from "chart.js/helpers";
+
+@Component({
+  selector: 'app-pie-loan',
+  templateUrl: './pie-loan.component.html',
+  styleUrls: ['./pie-loan.component.scss']
+})
+export class PieLoanComponent implements OnInit{
+@Input() loanInfo?: Observable<MonthlyInterest>;
```

Comment:
Use `AsyncPipe` in the parent component HTML and pass the `@Input` here as a plain value. Then you can use `ngOnChanges` instead of `ngOnInit` to just draw the chart without the subscription.

Reply:
Did it all by myself :)